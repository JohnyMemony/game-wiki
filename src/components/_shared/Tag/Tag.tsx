import React from 'react';
import { Badge } from 'evergreen-ui';

interface Props {
  text: string;
  color?: 'green' | 'automatic' | 'neutral' | 'blue' | 'red' | 'orange' | 'yellow' | 'teal' | 'purple';
}

export const Tag: React.FC<Props> = (props) => {
  const { text, color } = props;

  return (
    <Badge className="tag" color={color}>{text}</Badge>
  );
};
