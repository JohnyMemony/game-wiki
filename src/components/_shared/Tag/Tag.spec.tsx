import React from 'react';
import { shallow } from 'enzyme';
import { Tag } from './Tag';

describe('Tag component', () => {
  it('should render Tag component', function () {
    const component = shallow(<Tag text="Tag text" />);
    expect(component).toMatchSnapshot();
  });
});
