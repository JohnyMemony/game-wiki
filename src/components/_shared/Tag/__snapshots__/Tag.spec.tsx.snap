// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag component should render Tag component 1`] = `
<Memo(ForwardRef(Badge))
  className="tag"
>
  Tag text
</Memo(ForwardRef(Badge))>
`;
